rl_chain/rl_chain_base.py

```diff
@@ -215,18 +219,40 @@ def embed_string_type(item: Union[str, _Embed], model: Any, namespace: Optional[
     
     return {namespace: join_char.join(map(str, encoded))}
 
-def embed_dict_type(item: Dict, model: Any) -> Dict[str, str]:
+def embed_dict_type(item: Dict, model: Any) -> Dict[str, Union[str,List[str]]]:
     """Helper function to embed a dictionary item."""
     inner_dict = {}
-    for ns, embed_str in item.items():
-        inner_dict.update(embed_string_type(embed_str, model, ns))
+    for ns, embed_item in item.items():
+        if is_stringtype_instance(embed_item):
+            inner_dict.update(embed_string_type(embed_item, model, ns))
+        elif isinstance(embed_item, list):
+            inner_dict[ns] = []
+            for embed_list_item in embed_item:
+                if is_stringtype_instance(embed_list_item):
+                    embedded = embed_string_type(embed_list_item, model, ns)
+                    inner_dict[ns].append(embedded[ns])
+                else:
+                    raise ValueError(f"Unsupported type {type(embed_list_item)} for embedding.")
+        else:
+            raise ValueError(f"Unsupported type {type(embed_item)} for embedding.")
```

Comment:
if we move value type checking/raising to embed_string function, we can remove it from everywhere else

Reply:
true done